src/ApiClient.php

```diff
@@ -228,17 +305,34 @@ private function fetchQuotes(array $symbols)
     {
         $qs = $this->getRandomQueryServer();
 
-        // Initialize session cookies
-        $cookieJar = $this->getCookies();
+        $crumb = $this->getCrumb();
 
-        // Get crumb value
-        $crumb = $this->getCrumb($qs, $cookieJar);
+        $url = 'https://query'.$qs.'.finance.yahoo.com/v7/finance/quote?symbols='.urlencode(implode(',', $symbols)).'&crumb='.$crumb;
 
-        // Fetch quotes
-        $url = 'https://query'.$qs.'.finance.yahoo.com/v7/finance/quote?crumb='.$crumb.'&symbols='.urlencode(implode(',', $symbols));
-        $responseBody = (string) $this->client->request('GET', $url, ['cookies' => $cookieJar, 'headers' => $this->getHeaders()])->getBody();
+        try {
+            $responseBody = (string) $this->client->request('GET', $url, [
+                'cookies' => $this->getCookies(),
+                'headers' => $this->getHeaders(),
+            ])->getBody();
+
+            return $this->resultDecoder->transformQuotes($responseBody);
+        } catch (\GuzzleHttp\Exception\ClientException $e) {
+            // Retry once if Unauthorized
+            if (401 === $e->getResponse()->getStatusCode()) {
+                $this->refreshCookiesAndCrumb();
+                $crumb = $this->getCrumb();
+
+                $url = 'https://query'.$qs.'.finance.yahoo.com/v7/finance/quote?symbols='.urlencode(implode(',', $symbols)).'&crumb='.$crumb;
+                $responseBody = (string) $this->client->request('GET', $url, [
+                    'cookies' => $this->getCookies(),
+                    'headers' => $this->getHeaders(),
+                ])->getBody();
```

Comment:
If you wrap the try-catch in a for loop, you can avoid repeating this code block and have as many retries as you like:

```
for ($retries = 0; ++$retries; $retries < 1) {
    try {
        // Execute request
        break; // Break the loop when request was successful
     } catch (...) {
          // Do error handling
     }
}